src/wallet.ts

```diff
@@ -464,7 +464,13 @@ WalletMiddlewareOptions): JsonRpcMiddleware<any, Block> {
  */
 function validateVerifyingContract(data: string) {
   const { domain: { verifyingContract } = {} } = parseTypedMessage(data);
-  if (verifyingContract && !isValidHexAddress(verifyingContract)) {
+  // Explicit check for cosmos here has been added to address this issue
+  // https://github.com/MetaMask/metamask-extension/issues/26980
+  if (
+    verifyingContract &&
+    (verifyingContract as string) !== 'cosmos' &&
```

Comment:
We're going to introduce this single exception? Doesn't it seem possible there are other cases here? Do we have a plan to deprecate support for this exception and start enforcing this more strictly or? 

cc @MetaMask/wallet-api-platform-engineers

Reply:
Cosmos network DAPPs are using this deprecated SDK which has verifyingContract hard coded to `cosmos` https://github.com/evmos/ethermint/blob/main/ethereum/eip712/domain.go#L29 thus we need to do it to support cosmos network. Once this old library goes out of use we can revert the change.

We do not see more such cases currently.

Reply:
if I understand correctly we are planning on removing this sometime in the future: if so, to make sure we don't lose track of it, perhaps we can open a new issue and mention it in the code comment instead of using the issue that will be closed by this PR (#26980)?

Reply:
Good point @mikesposito , I added this issue https://github.com/MetaMask/eth-json-rpc-middleware/issues/337 and updated the comment.